design/ceph-config-updates.md

```diff
@@ -0,0 +1,256 @@
+- `debug ...` configs
+
+Removed configs:
+- `mon_max_pg_per_osd = 1000` is a dangerous setting and should be removed regardless of whether
+  this proposal is accepted
+- `log file = /dev/stderr` is set by default to keep with container standards and kept here if the
+  user needs to change this for debugging/testing
+- `mon cluster log file = /dev/stderr` for `log file` reasons above
+- `mon keyvaluedb = rocksdb` is not needed for Luminous+ clusters
+- `filestore_omap_backend = rocksdb` is not needed for Luminous+
+- `osd pg bits = 11` set (if needed) using config override for testing or play clusters
+- `osd pgp bits = 11` set (if needed) using config override for testing or play clusters
+- `osd pool default size = 1` set (if needed) using config override for testing or play clusters
+- `osd pool default min size = 1` set (if needed) using config override for testing or play clusters
+- `osd pool default pg num = 100` set (if needed) using config override for testing or play clusters
+- `osd pool default pgp num = 100` set (if needed) using config override for testing or play clusters
+- `rbd_default_features = 3` kubernetes should support Ceph's default RBD features after k8s v1.8
+
+#### Additional configs via user override
+
+Rook currently offers the option of a config override in a ConfigMap which users may modify after
+the Ceph operator has started. We propose to keep the "spirit" of this functionality but change the
+method of implementation, as the ConfigMap modification approach will be hard to integrate with the
+final goal of eliminating the config file altogether. Instead, we propose to update the Ceph cluster
+CRD to support setting and/or overriding values at the time of cluster creation. The proposed format
+is below.
+
+```yaml
+apiVersion: ceph.rook.io/v2alpha1
+kind: CephCluster
+spec:
+  # ...
+  # For advanced users:
+  # 'config' adds or overrides values in the Ceph config at operator start time and when the cluster
+  # CRD is updated. Config changes are made in the mon's centralized config if it is available
+  # (Mimic+) so that the user may override them temporarily via Ceph's command line. For Luminous,
+  # the changes are set on the command line since the centralized config is not available, and
+  # temporary overrides will not be possible.
+  config:
+    # Each key in the 'config' section represents a config file section. 'global' is likely to be
+    # the only section which is modified; however, daemons can have their config overridden
+    # explicitly if desired.
+    # global will add/override config for all Ceph daemons
+    global:
+      # The below "osd_pool_default..." settings make the default pools created have no replication
+      # and should be removed for production clusters, as this could impact data fault tolerance.
+      osd_pool_default_size: "1",
+      osd_pool_default_min_size: "1",
+    # mon. will add/override config for all mons
+    mon.:
```

Comment:
s/mon./mon/ to be consistent with the way ceph names the sections (in teh conf file or via the new cli and mon database)